ReactWindows/ReactNative.Shared/Modules/Core/IReactChoreographer.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Portions derived from React Native:
+// Copyright (c) 2015-present, Facebook, Inc.
+// Licensed under the MIT License.
+
+using System;
+
+namespace ReactNative.Modules.Core
+{
+    /// <summary>
+    /// A simple action queue that allows us to control the order certain
+    /// callbacks are executed within a given frame.
+    /// </summary>
+    public interface IReactChoreographer : IDisposable
+    {
+        /// <summary>
+        /// For use by <see cref="UIManager.UIManagerModule"/>. 
+        /// </summary>
+        event EventHandler<FrameEventArgs> DispatchUICallback;
+
+        /// <summary>
+        /// For use by <see cref="Animated.NativeAnimatedModule"/>. 
+        /// </summary>
+        event EventHandler<FrameEventArgs> NativeAnimatedCallback;
+
+        /// <summary>
+        /// For events that make JavaScript do things.
+        /// </summary>
+        event EventHandler<FrameEventArgs> JavaScriptEventsCallback;
+
+        /// <summary>
+        /// Event used to trigger the idle callback. Called after all UI work has been
+        /// dispatched to JavaScript.
+        /// </summary>
+        event EventHandler<FrameEventArgs> IdleCallback;
```

Comment:
>FrameEventArgs [](start = 27, length = 14)

I suspect we'll want to ensure FrameEventArgs isn't sealed or use an interface. Doesn't have to be done now, but certainly once we move it to a separate .csproj.

Reply:
I think an interface is the best approach. I'm not sure how people typically ship standalone interfaces with event args. I don't know if I've seen it done, so can't comment on "the right thing to do" here.

---
In reply to: [181599952](https://github.com/Microsoft/react-native-windows/pull/1773#discussion_r181599952) [](ancestors = 181599952)

Reply:
I can bring over the IFrameEventArgs from the other branch, but I’m not sure if you’re sure that’s what you want? ;) Having DTOs with primitives isn’t unheard of in interface-only libraries. I do understand not wanting the class to be sealed

Reply:
For reference from RxJS, they use parameterized types to inject arbitrary EventArgs:
https://github.com/Reactive-Extensions/Rx.NET/blob/develop/Rx.NET/Source/src/System.Reactive/IEventPattern.cs
https://github.com/Reactive-Extensions/Rx.NET/search?utf8=%E2%9C%93&q=EventArgs&type=

This does mean there's more density to method declarations, but thanks to 'var' and contravariance/covariance the impact on the developer experience for RN .NET framework and module developers should be minimal.